rdkcentral/Lightning-UI-Components
# fix(Tile): fixes transitions on metadata metadata

## Description

This PR does the following:

- fixes metadataPatch so bottom animation works  every time Tile goes from unfocus to focus
- restructures the code, breaking down components into sections for readability
- adds more logic to the shouldShowMetadata to account for all possible scenarios instead of just persistentMetadata and isFocused

## References

LUI-658

## Testing

- pull down branch
- `yarn start`
-  go to Tile story
- select `inset` in controls
- click `focus` in controls, bottom metadata data should move from bottom to top
- click `unfocus` and `focus` again the transition should perform as expected

## Automation



## Checklist
